shim/shim.c

```diff
@@ -0,0 +1,399 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <poll.h>
+#include <assert.h>
+#include <stdarg.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <stdint.h>
+#include <inttypes.h>
+
+/* The shim would be handling fixed number of predefined fds.
+ * This would be signal fd, stdin fd, proxy socket fd and an I/O
+ * fd passed by the runtime
+ */
+#define MAX_POLL_FDS 4
+
+/* These are the command ids used by hyperstart control message format.
+ * cc-shim sends WINSIZE command on receving a SIGWINCH signal and 
+ * a KILLCONTAINER command on receiving any other signal
+ */
```

Comment:
would be better to have a struct or a enum that maps all hyper commands in a special file like  hyper_agent.c